model/task/db.go

```diff
@@ -2640,12 +2640,14 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 // fetching tasks to block them, and for each task, all dependencies
 // that have a status that that would block the task from running (i.e., it is
 // inconsistent with the task's Dependency.Status field) and have not been marked
-// unattainable will be retrieved.
+// unattainable will be retrieved. The updateAllDependencies indicates whether
+// all tasks that depend on the given tasks will be updated (i.e., whether
+// the task's Dependency.Status will be ignored).
 //
 // This must find tasks in smaller chunks to avoid the 16 MB query size limit -
 // if the number of tasks is large, a single query could be too large and the DB
 // will reject it.
-func FindAllDependencyTasksToModify(tasks []Task, isUnblocking bool) ([]Task, error) {
+func FindAllDependencyTasksToModify(tasks []Task, isUnblocking, updateAllDependencies bool) ([]Task, error) {
```

Comment:
nit: might be a little clearer if we called this `ignoreDependencyStatusForBlocking` or something since it's doing something quite specific and whether or not it has any effect depends on `isUnblocking`.

Reply:
nit: since `updateAllDependencies` is for blocking only (i.e. if it's true, then it has an effect when it's blocking) and `isUnblocking` is stating the opposite (if it's true, then it's unblocking), could we flip the boolean to be `isBlocking` instead of `isUnblocking`? Thinking through the logic is a little easier if we don't have one boolean indicating the negative and one indicating the positive.